feat(react-scheduler): add appointment resources support (#2498)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/resources.md

# React Scheduler - Resources

The React Scheduler allows you to assign appointments to different resources. For example, if the Scheduler illustrates talks at a conference, resources could be rooms, speakers, and attendees. 

In the UI, resources are identified by different colors. If the colors are not defined, they are assigned automatically.

To assign appointments to resources, users can edit a dedicated [AppointmentForm](../reference/appointment-form.md) field.

## Related Plugins

- [Resources](../reference/resources.md) - configures resources

## Single and Multiple Instance Resources

> Appointments can be assigned to multiple or only one instance of a resource, depending on the resource type. To illustrate the difference, let us return to the example in which appointments are talks at a conference and resources are rooms, speakers, and attendees.

Reviewer: ```suggestion
Appointments can be assigned to single/multiple resource instances, depending on the resource type. The same example about a conference can be used to illustrate the difference:
```

> A talk can only be held in a single room, so the appointment can only be assigned to one instance of the Rooms resource. Such resources are called **single instance resources**. This is the default resource type.

Reviewer: ```suggestion
A talk can only be held in one room, and the appointment can only be assigned to one instance of the Rooms resource. These are default resource types called **single instance resources**.
```
